net/net_test.go

```diff
@@ -21,6 +21,185 @@ import (
 	"testing"
 )
 
+func TestParseCIDRs(t *testing.T) {
+	testCases := []struct {
+		cidrs         []string
+		errString     string
+		errorExpected bool
+	}{
+		{
+			cidrs:         []string{},
+			errString:     "should not return an error for an empty slice",
+			errorExpected: false,
+		},
+		{
+			cidrs:         []string{"10.0.0.0/8", "not-a-valid-cidr", "2000::/10"},
+			errString:     "should return error for bad cidr",
+			errorExpected: true,
+		},
+		{
+			cidrs:         []string{"10.0.0.0/8", "2000::/10"},
+			errString:     "should not return error for good  cidrs",
+			errorExpected: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		cidrs, err := ParseCIDRs(tc.cidrs)
+		if tc.errorExpected {
```

Comment:
If you make errString an actual error and get rid of `errorExpected` you can do something like this instead:
```go
if reflect.DeepEqual(tc.expectedErr, err) {
   t.Errorf("...")
}

if tc.expectedErr != nil {
     return
}

if len(cidrs) != len(tc.cidrs) {
    t.Errorf(...)
}
```

Reply:
I honestly don't think we should test error messages. We can test error types.. but we are not using them here.

Reply:
Oh I see now.. errString is not the returned error from ParseCIDRsStrings but the error representing how the test would fail. I think that's sort of confusing, seems like that should just be in the test name/description